src/bda/plone/orders/common.py

```diff
@@ -521,13 +532,39 @@ def salaried(self, value):
             'Abstract OrderState does not implement salaried.setter')
 
     def update_item_stock(self, booking, old_state, new_state):
+        """Change stock according to transition. See table in transitions.py
+        """
         if old_state == new_state:
             return
         # XXX: fix stock item available??
-        if new_state == ifaces.STATE_NEW:
-            self.decrease_stock(booking)
-        if new_state == ifaces.STATE_CANCELLED:
-            self.increase_stock(booking)
+        if old_state == ifaces.STATE_NEW:
+            if new_state == ifaces.STATE_CANCELLED:
+                self.increase_stock(booking)
+            else:
```

Comment:
Yes, new bookings do decrease the stock by it's own, not via the ``update_item_stock`` method. So for new bookings, the stock is correctly changed.
https://github.com/bluedynamics/bda.plone.orders/blob/thet-reserveditems/src/bda/plone/orders/common.py#L415